app/src/main/java/com/vero/tasky/auth/util/EmailErrorHandler.kt

```diff
@@ -0,0 +1,19 @@
+package com.vero.tasky.auth.util
+
+import com.vero.tasky.R
+
+object EmailErrorHandler {
+
+    fun validateEmail(
+        isEmailValid: Boolean,
+        onEmailIsValid: () -> Unit,
+        showError: (message: Int) -> Unit
+    ) {
```

Comment:
Instead of passing tons of callbacks, I'd recommend to return a result class instead. Something like this:
```
data class ValidationResult(
    val isValid: Boolean,
    val error: Int?
)
```